### PR TITLE
transport: add git-send-pack support to local/ssh.

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

File: plumbing/protocol/packp/report_status.go [hunk 1]
```diff
@@ -95,6 +95,10 @@ func (s *ReportStatus) decodeReportStatus(b []byte) error {
 		return fmt.Errorf("premature flush")
 	}
 
+	if b[len(b)-1] == '\n' {
+		b = b[:len(b)-1]
+	}
+
```
Comment (alcortesm) [inline plumbing/protocol/packp/report_status.go]: bytes.TrimSuffix?

File: plumbing/protocol/packp/report_status.go [hunk 2]
```diff
@@ -106,6 +110,10 @@ func (s *ReportStatus) decodeReportStatus(b []byte) error {
 }
 
 func (s *ReportStatus) decodeCommandStatus(b []byte) error {
+	if b[len(b)-1] == '\n' {
+		b = b[:len(b)-1]
+	}
+
```
Comment (alcortesm) [inline plumbing/protocol/packp/report_status.go]: bytes.TrimSuffix?

File: plumbing/protocol/packp/report_status.go (hunk 1 above)
Comment (smola) [inline plumbing/protocol/packp/report_status.go]: Done.

File: plumbing/protocol/packp/report_status.go (hunk 2 above)
Comment (smola) [inline plumbing/protocol/packp/report_status.go]: Done.

**Comment by smola** [comment]
This PR fails on Travis because it uses git 1.8.x. See https://github.com/src-d/go-git/pull/170 for the fix and merge it before this one.

**Pull request merged**